### PR TITLE
Runtime fields: update comments on script context

### PR DESCRIPTION
**Pull request by nik9000**
We had some super informative comments on the script context
configuration that we share for all of runtime fields. But they were out
of date. They assumed that runtime fields can only show up in the
mapping.

### CODE REVIEW

**Comment by nik9000** [comment]
@jimczi found these out of date comments while investigating how runtime fields work with the can match phase. Fun times.

File: x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java [hunk 1]
```diff
@@ -41,18 +41,21 @@
...
             100,
             timeValueMillis(0),
             /*
-             * Disable compilation rate limits for scripted fields so we
+             * Disable compilation rate limits for runtime fields so we
              * don't prevent mapping updates because we've performed too
-             * many recently. That'd just be lame.
+             * many recently. That'd just be lame. We also compile these
+             * scripts during search requests so this could totally be a
+             * source of runaway script compilations. We think folks we
```
Comment (romseygeek) [inline x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java:57]: s/we mostly/will mostly/

File: x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java [hunk 2]
```diff
@@ -41,18 +41,21 @@
...
             /*
-             * In an ideal world we wouldn't need the script cache at all
-             * because we have a hard reference to the script. The trouble
-             * is that we compile the scripts a few times when performing
-             * a mapping update. This is unfortunate, but we rely on the
-             * cache to speed this up.
+             * We rely on the script cache in two ways:
+             * 1. It caches the "heavy" part of mappings generated at runtime.
+             * 2. Mapping updates tend to try to compile the script twice. Not
+             *    for any good reason. They just do.
```
Comment (romseygeek) [inline x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java:47]: @javanna is in the process of fixing this! Or at least, starting to.

**Review by romseygeek — APPROVED** [review]
One nit, LGTM otherwise

**Review by jimczi — APPROVED** [review]
Thanks @nik9000 !

File: x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java (hunk 2 above)
Comment (nik9000) [reply x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractFieldScript.java:47]: Nice.

**Pull request merged**